hooks/src/index.js

```diff
@@ -142,7 +142,8 @@ export function useImperativeHandle(ref, createHandle, args) {
 
 function bindHandles(handles) {
 	handles.some(handle => {
-		if (handle.ref) handle.ref.current = handle.createHandle();
+		if (typeof handle.ref === 'function') handle.ref(handle.createHandle());
+		else if (handle.ref) handle.ref.current = handle.createHandle();
```

Comment:
FYI: This looks pretty similar to `applyRef` in core. There is potential for golfing in the PR where hooks are merged into core :tada: